### PR TITLE
Align `storefront create` with react-storefront turborepo

### PR DESCRIPTION
**Pull request by 2can**

### CODE REVIEW

**Review by lkostrowski — APPROVED** [review]

**Review by lkostrowski — CHANGES_REQUESTED** [review]
Tested locally. Server on :3000 starts, but it fails running 

```
storefront:dev: Module not found: Can't resolve '@saleor/ui-kit'
storefront:dev: > 1 | import { ChipButton } from "@saleor/ui-kit";
storefront:dev:   2 | import React from "react";
storefront:dev:   3 | 
storefront:dev:   4 | import { Box } from "../Box";
storefront:dev: 
```

**Comment by 2can** [comment]
@lkostrowski `ui-kit` build step added

File: src/cli/storefront/create.ts [hunk 1]
```diff
@@ -204,20 +204,21 @@ export const createStorefront = async (argv: Arguments<StoreCreate>) => {
...
       '\n'
     )
   );
 
+  await run('pnpm', ['build', '--filter=ui-kit'], { cwd: process.cwd() });
+
   await run(
     'pnpm',
-    ['next', 'dev', '--port', detectedPort.toString()],
+    ['dev', '--filter=storefront'],
```
Comment (lkostrowski) [inline src/cli/storefront/create.ts:221]: @mmiszy correct me if I'm wrong, but AFAIK turbo can/should be configured with proper deps. So if we run storefront, ui-kit should build automatically and listen to changes.

Also, this is detail of storefront app. I don't think we should replicate internal scripts structure in CLI. This is fragile and CLI will break each time some of these scripts change.

I hope we can run single Storefront script and Storefront can handle it's start.

(ps I don't know why do we start it, IMO it's only to press ctrl+c)

**Comment by lkostrowski** [comment]
Confirmed locally, storefront starts

File: src/cli/storefront/create.ts [hunk 2]
```diff
@@ -204,20 +204,21 @@ export const createStorefront = async (argv: Arguments<StoreCreate>) => {
...
     chalk(
       chalk.bold(
-        `\n  Starting server on 0.0.0.0:${detectedPort}, url: http://localhost:${detectedPort}`
+        '\n  Starting server on 0.0.0.0:3000, url: http://localhost:3000'
       ),
       '\n'
     )
   );
 
+  await run('pnpm', ['build', '--filter=ui-kit'], { cwd: process.cwd() });
```
Comment (mmiszy) [inline src/cli/storefront/create.ts:217]:
```suggestion
  await run('pnpm', ['build', '--filter=storefront^...'], { cwd: process.cwd() });
```

File: src/cli/storefront/create.ts (hunk 1 above)
Comment (mmiszy) [reply src/cli/storefront/create.ts:221]: > AFAIK turbo can/should be configured with proper deps. So if we run storefront, ui-kit should build automatically and listen to changes.

Yes, I've added a comment above
https://github.com/saleor/saleor-cli/pull/254/files#r939877484

**Review by mmiszy — CHANGES_REQUESTED** [review]
Let's change `--filter=ui-kit` to `--filter=storefront^...` so that it's more loosely coupled with the internal structure of the storefront and more futureproof.

File: src/cli/storefront/create.ts (hunk 2 above)
Comment (mmiszy) [reply src/cli/storefront/create.ts:217]: Rel: https://turborepo.org/docs/core-concepts/filtering#include-dependencies-of-matched-packages

Comment (lkostrowski) [reply src/cli/storefront/create.ts:217]: @mmiszy do we need to build storefront? Last cmd is `dev` so maybe just `pnpm run dev --filter=storefront^...`?

Comment (mmiszy) [reply src/cli/storefront/create.ts:217]: How is that different from what I suggested @lkostrowski ?

Comment (mmiszy) [reply src/cli/storefront/create.ts:217]: We need to build storefront's dependencies. We're not building storefront here.

Comment (lkostrowski) [reply src/cli/storefront/create.ts:217]: I was thinking about `storefront dev` triggering `ui-kit dev`, so when I edit ui-kit in dev mode, it gets automatically updated in the tree. And `storefront dev` can (in turbo config) mark that `ui-kit build` must happen before (hence no need to write this explicitely). At least thats how I was configuring turbo before

Comment (mmiszy) [reply src/cli/storefront/create.ts:217]: If we want to start the `dev` script for storefront and its dependencies we need to run `pnpm dev --filter=storefront...`

Comment (lkostrowski) [reply src/cli/storefront/create.ts:217]: Yup and this is probably something we should do @2can @zaiste

**Review by lkostrowski — APPROVED** [review]
Tested locally, works

**Review by mmiszy — APPROVED** [review]

**Review by zaiste — APPROVED** [review]

**Pull request merged**